posthog/api/utils.py

```diff
@@ -187,8 +194,13 @@ def get_project_id(data, request) -> Optional[int]:
     return None
 
 
-def get_data(request):
+def get_data(request, csp_report=None, is_csp_report_ingestion=False):
     data = None
+    # Making the checks here stricter to avoid affecting any existing pipeline flow
+    # is_csp_report_ingestion is only true if we got here through urls.get_csp_event
+    if is_csp_report_ingestion and csp_report:
+        return csp_report, None
```

Comment:
i think we could also move this up and not call get_data if we have the csp report, since we can then avoid this change completely

(no change == no weird capture bug by accident 🤣)

Reply:
Done, testing it right now